ucsb-cs156-w24/proj-courses-w24-5pm-4
# FEATURE: Add edit page for personal schedules

Changed file: 
- frontend/src/main/components/PersonalSchedules/PersonalScheduleForm.js
- frontend/src/main/components/Quarters/SingleQuarterDropdown.js
- frontend/src/main/pages/PersonalSchedules/PersonalSchedulesEditPage.js
- frontend/src/stories/pages/HomePage/PersonalSchedules/PersonalSchedulesEditPage.stories.js
- frontend/src/tests/components/PersonalSchedules/PersonalScheduleForm.test.js
- frontend/src/tests/pages/PersonalSchedules/PersonalSchedulesEditPage.test.js

Dokku deployment: https://proj-courses-cory-zhao-dev.dokku-08.cs.ucsb.edu

This PR adds the PersonalScheduleEditPage. This page can edit the fields of any PersonalSchedule. After clicking edit, it should display the correct contents of the PersonalSchedule in the form and when you edit any changes, the changes should be consistent on the frontend and backend. Furthermore, each personal schedule contains a CoursesTable that displays all the courses linked to the PersonalSchedule by psId. When you delete a course on the edit page, it should delete the course on the frontend and backend. 

How to test:
Note - with each step, look in swagger to see the contents are correctly filling the backend

1. Add 2 schedules and click edit on the schedules. Check the form to see if the fields are filled in with the correct contents.
2. Edit the form. Check that the form is edited in frontend and backend.
3. Add 2 courses (you should add courses that have no sections and are allowed with the quarter you chose). Have each course link to a separate schedule. Check each PersonalSchedule to see only the linked course is in the CourseTable. 
4. Delete a course from the table. Check to see that course is deleted in the frontend and backend.

<img width="1338" alt="Screenshot 2024-03-06 at 3 02 45 PM" src="https://github.com/ucsb-cs156-w24/proj-courses-w24-5pm-4/assets/91992519/d6247309-fc79-4eef-ba58-fb70c78c5675">

Closes #11
